### PR TITLE
feat: notion db loader

### PR DESCRIPTION
**Pull request by munkhorgil**
Basic Notion Database loader.

### CODE REVIEW

**Comment by munkhorgil** [comment]
Hey @jacoblee93, this one should be good to go.

**Comment by jacoblee93** [comment]
Thank you! Will take a look as soon as I can.

File: docs/docs/modules/indexes/document_loaders/examples/web_loaders/notiondb.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (jacoblee93) [inline docs/docs/modules/indexes/document_loaders/examples/web_loaders/notiondb.md:1]: Can you change this to an `.mdx` file and move the example code to the `examples/` directory? 

https://github.com/hwchase17/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md

File: langchain/src/document_loaders/web/notiondb.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  public integrationToken: string;
+
+  public databaseId: string;
+
+  private headers: Record<string, string> = {};
+
+  constructor({ integrationToken, databaseId }: NotionDBLoaderParams) {
+    super();
+
+    if (!integrationToken) {
```
Comment (jacoblee93) [inline langchain/src/document_loaders/web/notiondb.ts:41]: Would be nice to try to load this from an environment variable like `process.env.NOTION_INTEGRATION_TOKEN`:

https://github.com/hwchase17/langchainjs/pull/1396/files#diff-e709c029fd4e48f9d3c541ae2399d64278cfe86491d525cd2e9d9f146281f3a8R33

File: langchain/src/document_loaders/web/notiondb.ts [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    this.integrationToken = integrationToken;
+    this.databaseId = databaseId;
+
+    this.headers = {
+      Authorization: `Bearer ${this.integrationToken}`,
+      "Content-Type": "application/json",
+      // @see https://developers.notion.com/reference/versioning
+      "Notion-Version": "2022-06-28",
```
Comment (jacoblee93) [inline langchain/src/document_loaders/web/notiondb.ts:56]: Can we make this version string passable as well?

File: langchain/src/document_loaders/web/notiondb.ts [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  private async retrievePagesIds(): Promise<string[]> {
+    const url = `${NOTION_BASE_URL}/databases/${this.databaseId}/query`;
+
+    const pageIds: string[] = [];
+    const query: { page_size: number; start_cursor?: number } = {
+      page_size: 100,
+    };
+    let hasMore = true;
+
+    while (hasMore) {
```
Comment (jacoblee93) [inline langchain/src/document_loaders/web/notiondb.ts:80]: Maybe some configurable limit here?

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Looks great overall - would love to see an integration test if possible as I'm sure this will be a popular one!

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Thank you! Will merge with the next release.

**Comment by munkhorgil** [comment]
> Thank you! Will merge with the next release.

Thanks for the fixes!

**Pull request merged**